File: tsp-solver-frontend/src/app/model/request/AntColonyRequest.ts
import {City} from "../City";

export class AntColonyRequest {
  cities: City[]
  alpha: number;
  beta: number;
  evaporationRate: number;
  q: number;
  antFactor: number;
  randomCitySelection: number;
  maxIterations: number;

  constructor(cities: City[], alpha: number, beta: number, evaporationRate: number, q: number,
              antFactor: number, randomCitySelection: number, maxIterations: number) {
    this.cities = cities;
    this.alpha = alpha;
    this.beta = beta;
    this.evaporationRate = evaporationRate;
    this.q = q;
    this.antFactor = antFactor;
    this.randomCitySelection = randomCitySelection;
    this.maxIterations = maxIterations;
  }

}

File: src/main/java/pl/edu/pbs/request/tsp/AntColonyRequest.java
package pl.edu.pbs.request.tsp;

import lombok.Data;
import pl.edu.pbs.tsp.City;

import java.util.List;

@Data
public class AntColonyRequest {

    private List<City> cities;
    private double alpha;
    private double beta;
    private double evaporationRate;
    private double q;
    private double antFactor;
    private double randomCitySelection;
    private int maxIterations;

}